agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/LoggingConfigurator.java

```diff
@@ -0,0 +1,269 @@
+import ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy;
+import ch.qos.logback.core.util.FileSize;
+import com.microsoft.applicationinsights.agent.bootstrap.diagnostics.DiagnosticsHelper;
+import com.microsoft.applicationinsights.agent.bootstrap.diagnostics.etw.EtwAppender;
+import com.microsoft.applicationinsights.agent.bootstrap.diagnostics.log.ApplicationInsightsDiagnosticsLogFilter;
+import com.microsoft.applicationinsights.agent.bootstrap.diagnostics.log.ApplicationInsightsJsonLayout;
+import com.microsoft.applicationinsights.agent.bootstrap.diagnostics.log.MoshiJsonFormatter;
+import com.microsoft.applicationinsights.agent.internal.wasbootstrap.configuration.Configuration;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.Path;
+import java.util.Locale;
+
+import static org.slf4j.Logger.ROOT_LOGGER_NAME;
+
+class LoggingConfigurator {
+
+    private final LoggerContext loggerContext;
+
+    public Level level;
+    public String destination;
+
+    public Path filePath;
+    public int fileMaxSizeMb;
+    public int fileMaxHistory;
+
+    LoggingConfigurator(Configuration.SelfDiagnostics selfDiagnostics, Path agentPath) {
+        loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
+
+        level = getLevel(selfDiagnostics.level);
+        destination = selfDiagnostics.destination;
+        filePath = agentPath.resolveSibling(selfDiagnostics.file.path);
+        fileMaxSizeMb = selfDiagnostics.file.maxSizeMb;
+        fileMaxHistory = selfDiagnostics.file.maxHistory;
+    }
+
+    void configure() {
+        loggerContext.getLogger(ROOT_LOGGER_NAME).detachAndStopAllAppenders();
+
+        if (DiagnosticsHelper.useAppSvcRpIntegrationLogging()) {
+            configureAppSvcs();
+        } else if (destination == null || destination.equalsIgnoreCase("file+console")) {
+            configureFileAndConsole();
+        } else if (destination.equalsIgnoreCase("file")) {
+            configureFile();
+        } else if (destination.equalsIgnoreCase("console")) {
+            configureConsole();
+        } else {
+            throw new IllegalStateException("Unknown self-diagnostics destination: " + destination);
+        }
+    }
+
+    private void configureAppSvcs() {
+        Logger rootLogger = loggerContext.getLogger(ROOT_LOGGER_NAME);
+        rootLogger.addAppender(configureFileAppender());
+        rootLogger.addAppender(configureConsoleAppender());
+
+        // Diagnostics IPA log file location. Disabled by default.
+        String diagnosticsOutputDirectory = System.getenv(DiagnosticsHelper.APPLICATIONINSIGHTS_DIAGNOSTICS_OUTPUT_DIRECTORY);
+        if (diagnosticsOutputDirectory != null && !diagnosticsOutputDirectory.isEmpty()) {
+            rootLogger.addAppender(configureDiagnosticAppender(diagnosticsOutputDirectory));
+        }
+
+        if (DiagnosticsHelper.isOsWindows()) {
+            rootLogger.addAppender(configureEtwAppender());
+        }
+        // TODO what about linux?
```

Comment:
a good question.  can we create a work item to track this? i thought app services have something already.. will need to follow up with Jenny.

Reply:
I think this is already tracked here: https://msazure.visualstudio.com/One/_workitems/edit/9606582